LdapForNet/LdapEntry.cs

```diff
@@ -232,14 +233,15 @@ protected override string GetKeyForItem(DirectoryAttribute item)
     public class ModifyAttributeCollection : KeyedCollection<string, DirectoryModificationAttribute>
     {
         internal ModifyAttributeCollection()
+            : base(StringComparer.OrdinalIgnoreCase)
         {
         }
 
         public ICollection<string> AttributeNames => Dictionary.Keys;
 
         protected override string GetKeyForItem(DirectoryModificationAttribute item)
         {
-            return item.Name + item.LdapModOperation.ToString();
+            return item.Name;
```

Comment:
@jaredrsowers this is the change that I'm talking about.

Reply:
@JManou, yeah, I see it now. But what is the point of having the key be a calculated value? That would defeat any benefit of a KeyedCollection, because you don't know the key.

Reply:
@jaredrsowers I agree with you, there is no benefit using a keyed collection for the ModifyAttributeCollection but there is one for SearchResultCollection. We could avoid iterating through the list each time we want to get an attribute value from a search result.